exploration/combined_rp.md

```diff
@@ -0,0 +1,417 @@
+```
+
+```python
+df_combined = calculate_groups_rp(
+    df_combined, by=["pcode"], col_name="cdi", ascending=False
+)
+df_combined = calculate_groups_rp(
+    df_combined, by=["pcode"], col_name="seas5", ascending=True
+)
+total_years = df_combined["year"].nunique()
+```
+
+```python
+dff
+```
+
+```python
+dicts = []
+for cdi_rp in df_combined["cdi_rp"].unique():
+    dff = df_combined[df_combined["cdi_rp"] >= cdi_rp]
+    dicts.append(
+        {
+            "rp_cdi_ind": cdi_rp,
+            "rp_cdi_combined": (
+                (total_years + 1) / dff["year"].nunique()
+                if dff["year"].nunique() > 0
+                else np.inf
+            ),
+        }
+    )
+
+df_cdi_rps = pd.DataFrame(dicts)
+```
+
+```python
+fig, ax = plt.subplots(dpi=200, figsize=(6, 6))
+
+df_cdi_rps.sort_values("rp_cdi_ind").plot(
+    x="rp_cdi_ind", y="rp_cdi_combined", legend=False, ax=ax
+)
+
+ax.set_xlim([1, 43])
+ax.set_ylim([1, 43])
+
+ax.set_title("Window B (Apr CDI) RP")
+ax.set_xlabel("Individual province RP (years)")
+ax.set_ylabel("Combined Window B RP (years)")
+
+ax.spines["top"].set_visible(False)
+ax.spines["right"].set_visible(False)
+```
+
+```python
+dicts = []
+for seas5_rp in df_combined["seas5_rp"].unique():
+    dff = df_combined[df_combined["seas5_rp"] >= seas5_rp]
+    dicts.append(
+        {
+            "rp_seas5_ind": seas5_rp,
+            "rp_seas5_combined": (
+                (total_years + 1) / dff["year"].nunique()
+                if dff["year"].nunique() > 0
+                else np.inf
+            ),
+        }
+    )
+
+df_seas5_rps = pd.DataFrame(dicts)
+```
+
+```python
+fig, ax = plt.subplots(dpi=200, figsize=(6, 6))
+
+df_seas5_rps.sort_values("rp_seas5_ind").plot(
+    x="rp_seas5_ind", y="rp_seas5_combined", legend=False, ax=ax
+)
+
+ax.set_xlim([1, 43])
+ax.set_ylim([1, 43])
+
+ax.set_title("Window A (Feb SEAS5) RP")
+ax.set_xlabel("Individual province RP (years)")
+ax.set_ylabel("Combined Window A RP (years)")
+
+ax.spines["top"].set_visible(False)
+ax.spines["right"].set_visible(False)
+```
+
+```python
+dff
+```
+
+```python
+dicts = []
+for cdi_rp in df_combined["cdi_rp"].unique():
+    for seas5_rp in df_combined["seas5_rp"].unique():
+        dff = df_combined[
+            (df_combined["cdi_rp"] >= cdi_rp)
+            | (df_combined["seas5_rp"] >= seas5_rp)
+        ]
+        dicts.append(
+            {
+                "rp_cdi_ind": cdi_rp,
+                "rp_seas5_ind": seas5_rp,
+                "rp_overall": (
+                    (total_years + 1) / dff["year"].nunique()
+                    if dff["year"].nunique() > 0
+                    else np.inf
+                ),
+            }
+        )
+
+df_rps = pd.DataFrame(dicts)
+```
+
+```python
+df_rps = df_rps.merge(df_seas5_rps).merge(df_cdi_rps)
+```
+
+```python
+total_years
+```
+
+```python
+cols = ["rp_seas5_combined", "rp_cdi_combined"]
+df_rps_acceptable.sort_values(cols).drop_duplicates(subset=cols, keep="last")
+```
```

Comment:
chunk needs to be moved down a ways in script until after `df_rps_acceptable` data.frame is created